apis/placement/v1beta1/overriddenresourcesnapshot_types.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",shortName=rss,categories={fleet,fleet-placement}
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.metadata.generation`,name="Gen",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OverriddenClusterResourceSnapshot is used to store the result of ClusterResourceSnapshot after being overridden on the
+// target cluster.
+// Once the snapshot has been overridden by the system, the manifest won't be updated because of the cluster label change (e.g
+// the override rule are not apply to the target cluster anymore). To make the changes effective, you need to update the
+// resource or its related override.
+//
+// Its spec is immutable.
+// We may need to produce more than one overriddenClusterResourceSnapshot for all the resources a placement selected to
+// get around the 1MB size limit of k8s objects.
+// The naming convention of a overriddenClusterResourceSnapshot is {clusterResourceSnapshotName}-{targetCluster}-{random-suffix}.
+// Each snapshot MUST have the following labels:
+//   - `CRPTrackingLabel` which points to its owner CRP.
+//   - `MasterSnapshotNameLabel` which is the name of the master overriddenClusterResourceSnapshot in a group.
+//   - `ClusterResourceSnapshotNameLabel` which is the name of the clusterResourceSnapshot.
+//   - `TargetClusterLabel` which is the overridden result of the clusterResourceSnapshot on the target cluster.
+//   - `IsLatestSnapshotLabel` which indicates whether the snapshot is the latest one.
+//
+// All the snapshots within the same index group must have the same ResourceIndexLabel.
+type OverriddenClusterResourceSnapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of OverriddenResourceSnapshot.
+	// +required
+	Spec OverriddenResourceSnapshotSpec `json:"spec"`
+
+	// The observed status of ResourceSnapshot.
+	// +optional
+	Status OverriddenResourceSnapshotStatus `json:"status,omitempty"`
+}
+
+type OverriddenResourceSnapshotSpec struct {
+	TargetCluster string `json:"targetCluster"`
+
+	ClusterResourceSnapshotName string `json:"clusterResourceSnapshotName"`
+
+	ResourceOverrides []OverrideSnapshotIdentifier `json:"resourceOverrides"`
```

Comment:
Have we decided to not have priority for now? Are the elements in the array sorted by anything or just random?

Reply:
btw, What exactly is this field for?